src/modules/cookieHunter/cookieHunter.helpers.ts

```diff
@@ -0,0 +1,187 @@
+import { CronJob } from 'cron';
+import type {
+  ChatInputCommandInteraction,
+  Client,
+  Message,
+  MessageReaction,
+  PartialMessageReaction,
+  PartialUser,
+  User,
+} from 'discord.js';
+
+import { cache } from '../../core/cache';
+import { ONE_MINUTE } from '../../helpers/timeConstants';
+
+const IT_IS_SNACK_TIME = '0 30 16 * * *'; // 4:30pm every day
+
+let jobCurrentlyRunning: CronJob | null = null;
+
+const sendMessageInRandomChannel = async (client: Client<true>) => {
+  const channel = await cache.get('cookieHunterChannels', []);
+  if (!channel.length) return;
+
+  const randomChannel = channel[Math.floor(Math.random() * channel.length)];
+  if (!randomChannel) return;
+
+  const channelToSend = await client.channels.fetch(randomChannel);
+
+  if (!channelToSend || !channelToSend.isTextBased()) return;
+  const cookieMessage = await channelToSend.send('**👵 Qui veut des cookies ?**');
+  await cache.set('currentHuntMessageId', cookieMessage.id);
+  await cache.set('cookieHunterDailyCount', {});
+  await cookieMessage.react('🥛');
+  await cookieMessage.react('🍪'); // 1 point for grandma here, she beats everyone who doesn't find her
+
+  setTimeout(() => void dailyHuntEnd(client, cookieMessage), ONE_MINUTE);
```

Comment:
maybe overkill to manage this now, and i m not sure often the bot crash/restarts. but what happens if the bot crashes? the daily hunt will never stop

a simple fix would be to manage it in `countCookies` method. when some dude react, you check if the message was posted more than a minute ago and if so, you don't count the point and manually call `dailyHuntEnd`

Reply:
yes I'm not a big fan of this setTimeout too... I will need to store the start time to do that but look like a good solution to avoid an infinite hunt